docs-src/0.4/en/getting_started/mobile.md

```diff
@@ -145,13 +150,112 @@ fn app(cx: Scope) -> Element {
-                for item in items.iter() {
+                for item in items.read().iter() {
+                    div { "- {item}" }
+                }
+            }
+        }
+    }
+}
+```
+
+You complete `lib.rs` should now look something like this (with some slightly different names in the generated code):
+
+```rust
+use anyhow::Result;
+#[cfg(target_os = "android")]
+use wry::android_binding;
+
+#[cfg(target_os = "android")]
+fn init_logging() {
+    android_logger::init_once(
+        android_logger::Config::default()
+            .with_min_level(log::Level::Trace)
+            .with_tag("dioxus-mobile-test"), // This name may be different for your application
+    );
+}
+
+#[cfg(not(target_os = "android"))]
+fn init_logging() {
+    env_logger::init();
+}
+
+#[cfg(any(target_os = "android", target_os = "ios"))]
+fn stop_unwind<F: FnOnce() -> T, T>(f: F) -> T {
+    match std::panic::catch_unwind(std::panic::AssertUnwindSafe(f)) {
+        Ok(t) => t,
+        Err(err) => {
+            eprintln!("attempt to unwind out of `rust` with err: {:?}", err);
+            std::process::abort()
+        }
+    }
+}
+
+#[cfg(any(target_os = "android", target_os = "ios"))]
+fn _start_app() {
+    stop_unwind(|| main().unwrap());
+}
+
+#[no_mangle]
+#[inline(never)]
+#[cfg(any(target_os = "android", target_os = "ios"))]
+pub extern "C" fn start_app() {
+    #[cfg(target_os = "android")]
+    android_binding!(com_example, dioxus_mobile_test, _start_app); // This code may be different for your application
+    #[cfg(target_os = "ios")]
+    _start_app()
+}
+
+use dioxus::prelude::*;
+
+pub fn main() -> Result<()> {
+    // Right now we're going through dioxus-desktop but we'd like to go through dioxus-mobile
+    // That will seed the index.html with some fixes that prevent the page from scrolling/zooming etc
+    dioxus_desktop::launch_cfg(
+        app,
+        // Note that we have to disable the viewport goofiness of the browser.
+        // Dioxus_mobile should do this for us
+        dioxus_desktop::Config::default().with_custom_index(r#"<!DOCTYPE html>
+        <html>
+          <head>
+            <title>Dioxus app</title>
+            <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" />
+            <!-- CUSTOM HEAD -->
+          </head>
+          <body>
+            <div id="main"></div>
+            <!-- MODULE LOADER -->
+          </body>
+        </html>
+       "#.into()),
+    );
+
+    Ok(())
+}
+
+fn app(cx: Scope) -> Element {
+    let items = use_ref(cx, || vec![1, 2, 3]);
+
+    log::debug!("Hello from the app");
+
+    render! {
+        div {
+            h1 { "Hello, Mobile"}
+            div { margin_left: "auto", margin_right: "auto", width: "200px", padding: "10px", border: "1px solid black",
+                button {
+                    onclick: move|_| {
+                        println!("Clicked!");
+                        items.write().push(0);
```

Comment:
Consider changing this to
```rust
let len = items.read().len();
items.write().push(len+1);
```
In order to continue the count, rather than only adding 0's.